src/types.rs

```diff
@@ -475,3 +506,152 @@ impl<'buf> RecvMsgOut<'buf> {
         self.header.flags
     }
 }
+
+/// [CancelBuilder] constructs match criteria for request cancellation.
+///
+/// The default behavior of the [CancelBuilder] is cancel all in-flight requests,
+/// however this can be refined to cancel only requests which reference a file descriptor
+/// or have a specific `user_data` value.
+///
+/// Additionally, [CancelBuilder::all] can be set to match multiple requests.
+///
+/// ### Examples
+///
+/// ```
+/// use io_uring::types::{CancelBuilder, Fd, Fixed};
+///
+/// // Match all in-flight requests.
+/// CancelBuilder::new();
```

Comment:
This seems counter-intuitive, maybe it should be called `CancelBuilder::any()` ?

Reply:
`CancelBuilder::any()` sounds good to me. Should we likewise make the `.all()` method a constructor too? I don't think users will ever want to call `CancelBuilder::any().all()`.

Reply:
I think it would be nice to have `all()`.

Reply:
@quininer @FrankReh I changed the builder yet again, I think it might be a bit easier to use now. We allow users to do the following:

```
| op                                 | action                            |
|------------------------------------+-----------------------------------|
| CancelBuilder::any()               | Cancel all in-flight ops          |
| CancelBuilder::any().all()         | Cancel all in-flight ops          |
| CancelBuilder::user_data(42)       | Cancel the first op with udata=42 |
| CancelBuilder::user_data(42).all() | Cancel all ops with udata=42      |
| CancelBuilder::fd(1)               | Cancel the first op with fd=1     |
| CancelBuilder::fd(1).all()         | Cancel all ops with fd=1          |
```

Reply:
I don't have a strong opinion, I think it's all good. :)